aptos-move/block-executor/src/txn_provider/blocking_txns_provider.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::txn_provider::TxnProvider;
+use aptos_mvhashmap::types::TxnIndex;
+use aptos_types::transaction::BlockExecutableTransaction as Transaction;
+use std::sync::{Arc, Condvar, Mutex};
+
+#[allow(dead_code)]
+pub enum BlockingTransactionStatus<T: Transaction> {
+    Ready(Arc<T>),
+    Waiting,
+}
+
+pub struct BlockingTransaction<T: Transaction> {
+    pub txn: Mutex<BlockingTransactionStatus<T>>,
+    pub cvar: Condvar,
+}
+
+#[allow(dead_code)]
+impl<T: Transaction> BlockingTransaction<T> {
+    pub fn new() -> Self {
+        Self {
+            txn: Mutex::new(BlockingTransactionStatus::Waiting),
+            cvar: Condvar::new(),
+        }
+    }
+}
+
+pub struct BlockingTxnsProvider<T: Transaction> {
+    txns: Vec<BlockingTransaction<T>>,
```

Comment:
I think `BlockingTransaction<T>` can just be replaced by `OnceCell<Arc<T>>`. The semantics are the same and I'd expect `OnceCell` to be on par or maybe more performant than a mutex + cvar.

Reply:
Done, this actually makes the code simpler.
We can now use` fn get_txn(&self, idx: TxnIndex) -> &T `(instead of `fn get_txn(&self, idx: TxnIndex) -> Arc<T>`)

Reply:
nit: this is called BlockingTxnsProvider while the other (default) is DefaultTxnProvider, let's pick txns or txn for the name